File: backup/Twitch/app/src/main/java/com/exact/twitch/ui/videos/TopVideosFragment.java
package com.exact.twitch.ui.videos;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.View;

import com.exact.twitch.R;
import com.exact.twitch.ui.fragment.RadioButtonBottomSheetDialogFragment;
import com.exact.twitch.util.FragmentUtils;

public class TopVideosFragment extends BaseVideosFragment implements RadioButtonBottomSheetDialogFragment.OnOptionSelectedListener {

    private int[] sortOptions = new int[] { R.string.this_week, R.string.this_month, R.string.all_time };

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (isFragmentVisible()) {
            int defaultSortItem = sortOptions[0];
            if (getViewModel().getCurrentPeriodText().getValue() == null) {
                getViewModel().getCurrentPeriodText().postValue(getString(defaultSortItem));
            }
            if (getViewModel().getSelectedPeriodItem() == null) {
                getViewModel().setSelectedPeriodItem(defaultSortItem);
            }
            if (getViewModel().getSort() == null) {
                getViewModel().setSort("views");
            }
            getViewModel().getCurrentPeriodText().observe(this, currentSortOption::setText);
        }
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        sortByContainer.setOnClickListener(v -> FragmentUtils.INSTANCE.showRadioButtonDialogFragment(
                requireActivity(),
                getChildFragmentManager(),
                sortOptions,
                getViewModel().getSelectedPeriodItem())
        );
    }

    @Override
    public void onSelect(int optionId, CharSequence optionText) {
        if (getViewModel().getSelectedPeriodItem() != optionId) {
            String period = null;
            switch (optionId) {
                case R.string.this_week:
                    period = "week";
                    break;
                case R.string.this_month:
                    period = "month";
                    break;
                case R.string.all_time:
                    period = "all";
                    break;
            }
            getViewModel().setSelectedPeriodItem(optionId);
            getViewModel().setPeriod(period);
            getViewModel().getCurrentPeriodText().postValue(optionText);
            loadData(true);
        }
    }

    @Override
    public void loadData(boolean override) {
        getViewModel().loadVideos(null, getViewModel().getPeriod(), null, null, getViewModel().getSort(), override);
    }

    @Override
    protected void initData() {
        loadData(false);
    }
}

File: app/src/main/java/com/exact/twitch/ui/videos/TopVideosFragment.kt
package com.exact.twitch.ui.videos

import android.os.Bundle
import android.view.View

import com.exact.twitch.R
import com.exact.twitch.ui.fragment.RadioButtonDialogFragment
import com.exact.twitch.util.FragmentUtils
import kotlinx.android.synthetic.main.fragment_videos.*

class TopVideosFragment : BaseVideosFragment(), RadioButtonDialogFragment.OnOptionSelectedListener {

    private companion object {
        val sortOptions = listOf(R.string.this_week, R.string.this_month, R.string.all_time)
        const val DEFAULT_INDEX = 0
        const val TAG = "TopVideos"
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        sortByRl.setOnClickListener { FragmentUtils.showRadioButtonDialogFragment(requireActivity(), childFragmentManager, sortOptions, DEFAULT_INDEX, TAG) }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if (isFragmentVisible) {
            if (!viewModel.isInitialized()) {
                viewModel.sort = Sort.VIEWS
                viewModel.period = Period.WEEK
                viewModel.periodText.postValue(getString(sortOptions[DEFAULT_INDEX]))
            }
            initDefaultSortTextObserver()
        }
    }

    override fun loadData(override: Boolean) {
        viewModel.loadVideos(reload = override)
    }

    override fun onSelect(index: Int, text: CharSequence, tag: Int?) {
        if (viewModel.periodText != text) {
            viewModel.period = when(tag) {
                R.string.this_week -> Period.WEEK
                R.string.this_month -> Period.MONTH
                R.string.all_time -> Period.ALL
                else -> null
            }
            viewModel.periodText.postValue(text)
            loadData(true)
        }
    }
}
